Simulate Meteor Beam's SpA boost

Zrp200 opened the pull request.

lawauditswe commented:
> * I am unsure if Sheer Force suppresses the boost.

Sheer Force should not suppress the Meteor Beam boost, nor should it boost the damage output.

lawauditswe reviewed (commented):
> * I am unsure if Sheer Force suppresses the boost.

Sheer Force should not suppress the Meteor Beam boost, nor should it boost the damage output.

Zrp200 commented:
> > * I am unsure if Sheer Force suppresses the boost.
> 
> Sheer Force should not suppress the Meteor Beam boost, nor should it boost the damage output.

In that case it should be good to go! The entire pull request is one line, after all!

ShivaD173 commented:
Can you include Electro Shot in this as well?

Zrp200 commented:
@ShivaD173 implemented in [9992afc](https://github.com/smogon/damage-calc/pull/534/commits/9992afc24e4dfb9d4d863592f7b19bf2a74ccb5f), that was on my list of things I wanted to do, thanks for reminding me.

I also implemented interactions with the Simple and Contrary abilities while I was at it, and added tests to match.

This should be ready to merge in.

Zrp200 commented on calc/src/mechanics/gen789.ts:
```diff
@@ -81,6 +81,16 @@ export function calculateSMSSSV(
   checkIntrepidSword(defender, gen);
 
+  if (move.named('Meteor Beam', 'Electro Shot')) {
+    // note: this doesn't work with neutralizing gas currently due to being processed before it
```
Yeah this is correct. The comment was written before I realized that. testing contrary leaf storm vs ngas makes it clear enough. I'll remove the comment, but I'm on mobile, so it'll be its own commit.
